tflearn/layers/conv.py

```diff
@@ -276,6 +371,96 @@ def max_pool_2d(incoming, kernel_size, strides=None, padding='same',
     return inference
 
 
+def max_pool_3d(incoming, kernel_size, strides=1, padding='same',
+                name="MaxPool3D"):
+    """ Max Pooling 3D.
+
+    Input:
+        5-D Tensor [batch, depth, rows, cols, channels].
+
+    Output:
+        5-D Tensor [batch, pooled depth, pooled rows, pooled cols, in_channels].
+
+    Arguments:
+        incoming: `Tensor`. Incoming 5-D Layer.
+        kernel_size: 'int` or `list of int`. Pooling kernel size.Must have kernel_size[0] = kernel_size[1] = 1
+        strides: 'int` or `list of int`. Strides of conv operation.Must have strides[0] = strides[4] = 1.
+            Default: [1 1 1 1]
+        padding: `str` from `"same", "valid"`. Padding algo to use.
+            Default: 'same'.
+        name: A name for this layer (optional). Default: 'MaxPool3D'.
+
+    Attributes:
+        scope: `Scope`. This layer scope.
+
+    """
+    assert padding in ['same', 'valid', 'SAME', 'VALID'], \
+        "Padding must be same' or 'valid'"
+
+    input_shape = utils.get_incoming_shape(incoming)
+    assert len(input_shape) == 5, "Incoming Tensor shape must be 5-D"
+
+    kernel = utils.autoformat_kernel_3d(kernel_size)
+    strides = utils.autoformat_stride_3d(strides) 
+    padding = utils.autoformat_padding(padding)
+
+    with tf.name_scope(name) as scope:
+        inference = tf.nn.max_pool(incoming, kernel, strides, padding)
+
+        # Track activations.
+        tf.add_to_collection(tf.GraphKeys.ACTIVATIONS, inference)
+
+    # Add attributes to Tensor to easy access weights
+    inference.scope = scope
+
+    return inference
+
+
+def avg_pool_3d(incoming, kernel_size, strides=None, padding='same',
+                name="AvgPool3D"):
+    """ Average Pooling 3D.
+
+    Input:
+        5-D Tensor [batch, depth, rows, cols, channels].
+
+    Output:
+        5-D Tensor [batch, pooled depth, pooled rows, pooled cols, in_channels].
+
+    Arguments:
+        incoming: `Tensor`. Incoming 5-D Layer.
+        kernel_size: 'int` or `list of int`. Pooling kernel size.Must have kernel_size[0] = kernel_size[1] = 1
+        strides: 'int` or `list of int`. Strides of conv operation.Must have strides[0] = strides[4] = 1.
+            Default: [1 1 1 1]
+        padding: `str` from `"same", "valid"`. Padding algo to use.
+            Default: 'same'.
+        name: A name for this layer (optional). Default: 'AvgPool3D'.
+
+    Attributes:
+        scope: `Scope`. This layer scope.
+
+    """
+    assert padding in ['same', 'valid', 'SAME', 'VALID'], \
+        "Padding must be same' or 'valid'"
+
+    input_shape = utils.get_incoming_shape(incoming)
+    assert len(input_shape) == 4, "Incoming Tensor shape must be 4-D"
+
+    kernel = utils.autoformat_kernel_3d(kernel_size)
+    strides = utils.autoformat_stride_3d(strides) 
```

Comment:
It cannot be because for maxpool3d 
because kernel_size and strides have different conditions to satisfy.
[https://www.tensorflow.org/versions/r0.9/api_docs/python/nn.html#max_pool3d](url)